### PR TITLE
[HttpFoundation] add @throws to getPayload

### LINKED ISSUE #50490: [HttpFoundation] getPayload throws JsonException if request body is empty.
### Symfony version(s) affected

6.3.0

### Description

If I call `$request->getPayload()` it throws a `JsonException`.

What is the intended behavior?
In my opinion throwing an exception is fine, but `@throws Symfony\Component\HttpFoundation\Exception\JsonException` on the method would be nice.

### How to reproduce

Call getPayload on request object in a controller and make POST request with empty body.

### Possible Solution

Add `@throws Symfony\Component\HttpFoundation\Exception\JsonException`

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by wiseguy1394**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no
| Tickets       | Fix #50490
| License       | MIT

Fixes #50490

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -1508,6 +1508,8 @@ public function getContent(bool $asResource = false)
 
     /**
      * Gets the decoded form or json request body.
+     * 
+     * @throws JsonException
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Request.php]:
```suggestion
     * @throws JsonException When the body cannot be decoded to an array
```

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @wiseguy1394.

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 2]
```diff
@@ -1508,6 +1508,8 @@ public function getContent(bool $asResource = false)
 
     /**
      * Gets the decoded form or json request body.
+     * 
+     * @throws JsonException When the body cannot be decoded to an array
```
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Request.php:1512]: should be `RequestExceptionInterface` IMHO

Comment (wiseguy1394) [reply src/Symfony/Component/HttpFoundation/Request.php:1512]: Why not exact JsonException?

Comment (ro0NL) [reply src/Symfony/Component/HttpFoundation/Request.php:1512]: @wiseguy1394 catching the general case seems more future proof

but no strong opionion, i dont believe in `@throws` bookkeeping anyway :)

**Comment by nicolas-grekas** [comment]
Behavior changed in #50730 for empty contents.

**Pull request merged**